### PR TITLE
Split schema

### PR DESCRIPTION
**Pull request by myluki2000**
## Description of changes
Implement content service in a way that it allows distributed inheritance.
  
## How has this been tested?
Only API tests, maybe we also want more unit tests but the most important stuff is tested.

- [ ] automated unit test
- [x] automated integration test
- [ ] automated acceptance test
- [ ] manual, exploratory test
    
## Checklist before requesting a review
- [x] My code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My code fulfills all acceptance criteria
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] The test coverage (line and branch) is reasonably high, especially on Service classes and other classes with complex logic
- [ ] I have made corresponding changes to the documentation
- [ ] I have added explanation of architectural decision and rationales to [wiki/adr](https://github.com/IT-REX-Platform/wiki/tree/main/adr)
- [ ] I have updated the changes in the ticket description

## Checklist for reviewer
- [ ] The code works and does not throw errors
- [ ] The code is easy to understand and there are no confusing parts
- [ ] The code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] The code change accomplishes what it is supposed to do
- [ ] I cannot think of any use case in which the code does not behave as intended
- [ ] The added and existing tests reasonably cover the code change
- [ ] I cannot think of any test cases, input or edge cases that should be tested in addition
- [ ] Description of the change is included in the documentation

### CODE REVIEW

File: src/main/resources/graphql/content.graphqls [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    """
+    metadata: ContentMetadata!
+}
+
+# add more types here
+
+"""
+Type of the assessment
+"""
+enum AssessmentType {
```
Comment (myluki2000) [inline src/main/resources/graphql/content.graphqls:103]: Is this really what Assessment type is supposed to be? And not just something like "Quiz", "Flashcard" etc.?

Comment (PaulBredl) [reply src/main/resources/graphql/content.graphqls:103]: no, check domain model

**Comment by PaulBredl** [comment]
Tests laufen nicht durch, weil die aktuelle Test Konfiguration anscheinend nicht mit H2 klarkommt. Wenn wir wissen, ob PostGre oder H2, kann man das angehen. Entweder dann Docker config fixen oder die tests hier auf postgre umstellen

**Comment by PaulBredl** [comment]
Die aktuellen Tests sind eher nur die wichtigsten Fälle. DIe Queries würde ich dann erst im Detail testen, wenn filter implementiert sind. Die Mutations könnte man an der ein oder anderen Stelle genauer testen I guess
